drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -41,12 +41,16 @@ static inline void msg_init(struct device *dev, struct i2c_msg *msg,
 	LL_I2C_SetTransferRequest(i2c, transfer);
 	LL_I2C_SetTransferSize(i2c, len);
 
-	if ((flags & I2C_MSG_RESTART) == I2C_MSG_RESTART) {
+
+	/* Only activate auto end mode if this message doesn't prepend or follow a restart condition */
+	if (	((flags & I2C_MSG_RESTART) == I2C_MSG_RESTART)
+		||	(LL_I2C_IsActiveFlag_TC(i2c))	)	{
```

Comment:
Why do you check the transfer completed flag here? The transfer hasn't started yet.

Reply:
Yes thats right for the first message of a transfer: 
TC not set, RESTART set or not set, so auto end mode will only be disabled when RESTART is set.

But for messages following a non auto end message (previous message had RESTART set):
TC is set -> This message needs to be ended manually. If we would activate  auto end mode here, it would lead to an immediate Stop Condition (because previous message is done, but waiting for a stop or restart condition)

Reply:
>But for messages following a non auto end message (previous message had RESTART set):
TC is set -> This message needs to be ended manually. If we would activate auto end mode here, it would lead to an immediate Stop Condition (because previous message is done, but waiting for a stop or restart condition)

Even if we don't enable auto end mode, an early stop condition may be generated in msg_done, as the TC flag is set and the while will evaluate to false. We could clear it inside the if but it turns out to be an ugly hack and make hard to follow the code. I believe we should not use auto end mode and handle it in software.

Reply:
I'd be OK with not using auto end mode. Certainly will make the Code easier to read.